### DISCUSSION TITLE
Needed FormItem Rule that triggers only on Form.validate()

### DISCUSSION THREAD

**Discussion by mPyth** [element-plus/element-plus#8666] [Q&A]
Is it possible to define required FormItem Rule that does NOT triggered on 'blur' or 'change', but only when Form.validate() is called ?

Here is [simple form](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8RWxGb3JtXG4gICAgcmVmPVwiZm9ybUVsXCJcbiAgICA6bW9kZWw9XCJyRm9ybUZpZWxkc1wiXG4gICAgOnJ1bGVzPVwiclZhbGlkUnVsZXNcIlxuICAgIGxhYmVsLXdpZHRoPVwiODBweFwiXG4gICAgQHN1Ym1pdC5wcmV2ZW50XG4gICAgc3R5bGU9XCJ3aWR0aDogZml0LWNvbnRlbnQ7IG1heC13aWR0aDogNDYwcHg7XCJcbiAgICA+XG5cdFx0PEVsRm9ybUl0ZW0gbGFiZWw9XCJVc2VybmFtZVwiIHByb3A9XCJ1bmFtZVwiPlxuICAgICAgPEVsSW5wdXQgdi1tb2RlbD1cInJGb3JtRmllbGRzLnVuYW1lXCI+PC9FbElucHV0PlxuICAgIDwvRWxGb3JtSXRlbT5cbiAgICBcblx0XHQ8RWxGb3JtSXRlbSBsYWJlbD1cIlBhc3N3b3JkXCIgcHJvcD1cInBhc3NcIj5cbiAgICAgIDxFbElucHV0IHYtbW9kZWw9XCJyRm9ybUZpZWxkcy5wYXNzXCI+PC9FbElucHV0PlxuICAgIDwvRWxGb3JtSXRlbT5cblxuICAgIDxFbEZvcm1JdGVtPlxuICAgICAgPGRpdiBAY2xpY2s9XCJvbkZvcmdvdFBhc3NcIiBzdHlsZT1cInRleHQtZGVjb3JhdGlvbjogdW5kZXJsaW5lOyBjb2xvcjogdmFyKC0tZWwtY29sb3ItcHJpbWFyeSk7IGZvbnQtc2l6ZTogdmFyKC0tZWwtZm9udC1zaXplLXNtYWxsKTsgdXNlci1zZWxlY3Q6IG5vbmU7IGN1cnNvcjogcG9pbnRlcjtcIj5Gb3Jnb3QgUGFzc3dvcmRcbiAgICAgIDwvZGl2PlxuICAgIDwvRWxGb3JtSXRlbT5cbiAgICBcbiAgICA8RWxGb3JtSXRlbSBzdHlsZT1cIm1hcmdpbi10b3A6IDMwcHg7XCI+XG4gICAgICA8RWxCdXR0b24gdHlwZT1cInByaW1hcnlcIiBAY2xpY2sucHJldmVudD1cIm9uU3VibWl0XCI+U2lnbiBJbjwvRWxCdXR0b24+XG4gICAgPC9FbEZvcm1JdGVtPlxuXHRcdFxuICA8L0VsRm9ybT5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYsIHJlYWN0aXZlIH0gZnJvbSAndnVlJztcblxuY29uc3QgZm9ybUVsID0gcmVmPEZvcm1JbnN0YW5jZSB8IG51bGw+KG51bGwpOyAgXG5cbmNvbnN0IHJGb3JtRmllbGRzID0gcmVhY3RpdmUoe1xuICAgIHVuYW1lOiAnJ1xuICAsIHBhc3M6ICcnXG59KTtcbiAgXG5jb25zdCByVmFsaWRSdWxlcyA9IHJlYWN0aXZlPEZvcm1SdWxlcz4oXG4gIHtcbiAgICAgIHVuYW1lOiBbeyByZXF1aXJlZDogdHJ1ZSwgbWVzc2FnZTogJ3BsZWFzZSBpbnB1dCBVc2VybmFtZScsIC8qdHJpZ2dlcjogJ2JsdXInKi8gfV1cbiAgICAsIHBhc3M6ICBbeyByZXF1aXJlZDogdHJ1ZSwgbWVzc2FnZTogJ3BsZWFzZSBpbnB1dCBQYXNzd29yZCcsIC8qdHJpZ2dlcjogJ2JsdXInKi8gfV1cbiAgfSk7XG5cbmZ1bmN0aW9uIG9uU3VibWl0KCk6IHZvaWRcbntcbiAgaWYgKCEgZm9ybUVsLnZhbHVlKVxuICBcdHJldHVybjtcbiAgXG4gIC8vY2xlYXJGb3JtRXJyTXNncygpO1xuICBmb3JtRWwudmFsdWUudmFsaWRhdGUoIChpc0Zvcm1WYWxpZDogYm9vbGVhbiwgaW52YWxpZEZpZWxkcz86IFZhbGlkYXRlRmllbGRzRXJyb3IpID0+XG4gICAge1xuICAgIFx0aWYgKCEgaXNGb3JtVmFsaWQpXG4gICAgXHR7XG4gICAgICBcdGNvbnNvbGUubG9nKCdmb3JtIGlzIG5vdCB2YWxpZCcpO1xuICAgICAgXHRyZXR1cm47XG4gICAgXHR9XG4gICAgICBcbiAgICAgIC8vIHN1Ym1pdCBjb2RlIGJlZ2lucyBoZXJlIC4uLlxuICBcdH0pO1xufVxuXG5mdW5jdGlvbiBvbkZvcmdvdFBhc3MoKTogdm9pZFxue1xuICAvLyBoZXJlIGlzIGNvZGUgZm9yIGNoYW5naW5nIGN1cnJlbnQgVmlld1xufVxuICBcbjwvc2NyaXB0PlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319) for demo purposes. If you have a solution feel free to change this demo.

I've tried next things:
\* to define a Rule without 'trigger'
\* to define a Rule whose 'trigger' is empty string
but seems that FormItem validation in all those cases is triggered as trigger='blur'.

Exactly the same question is asked in #2528, but no one answered.

**Comment by vasylpylypets** [comment]
```
<el-button type="primary" @click="submitForm(ruleFormRef)">Create</el-button>

const submitForm = (formEl) => {
    if (!formEl) return;
    formEl.validateField([], valid => {
        if (valid) {
            console.log('submit!');
        } else {
            console.log('error submit!', fields);
        }
    });
};
```

formEl.validateField(arrayOfFields, callback) - this method triggers validation
arrayOfFields - specific list of fields to validate or empty array as "all fields"

**Reply by mPyth** [reply]
Thanks for the suggestion, but I have impression your proposal (with specified non-empty array) will do partial validation when user clicks on a button (only those fields which are specified in array), but I need another functionality - to avoid validation when field lose focus.

**Comment by chenxch** [comment]
#2528
